packages/driver/src/cypress/utils.ts

```diff
@@ -403,28 +403,4 @@ export default {
   isPromiseLike (ret) {
     return ret && _.isFunction(ret.then)
   },
-
```

Comment:
Moved this function onto `cy` directly, because we want to expose it to libraries / users that want to write custom commands.

Reply:
Did we decided on the name we will use for the public API? I know we've got the concept of a `Chainable` in the public API, but would we be naming this `getSubjectFromChain` or something more concise like `cy.getSubject()`?

Reply:
So in this version there are a couple of methods on `cy`:

- `cy.subject()` - Resolves "what is the current subject" (optionally, for a chainerId other than the current one, but by default, just 'what's the subject right now').
- `cy.subjectChain()` - Returns the current subject chain (optionally, for a chainerId other than the current one).
- `cy.getSubjectFromChain()` - Takes a subject chain and resolves it to an actual value. No default behavior; you need a subjectChain to call this.

https://github.com/cypress-io/cypress/pull/24417/files#r1012215289 talks a little bit about why we need all three - but I think these names are both descriptive and short.

Open to suggestions if people have thoughts, but all three of them are needed, and I think each should be named with the other two in mind.

Reply:
These names seem reasonable. Considering the only real consumers of these APIs are mostly us, and "power users", I think it makes sense they use the standard Cypress nomenclature.